test/unit/org/apache/cassandra/io/sstable/LegacySSTableTest.java

```diff
@@ -637,18 +637,18 @@ public void testGenerateSstables() throws Throwable
 
         File ksDir = new File(LEGACY_SSTABLE_ROOT, String.format("%s/legacy_tables", BigFormat.latestVersion));
         ksDir.tryCreateDirectories();
-        copySstablesFromTestData(String.format("legacy_%s_simple", BigFormat.latestVersion), ksDir);
-        copySstablesFromTestData(String.format("legacy_%s_simple_counter", BigFormat.latestVersion), ksDir);
-        copySstablesFromTestData(String.format("legacy_%s_clust", BigFormat.latestVersion), ksDir);
-        copySstablesFromTestData(String.format("legacy_%s_clust_counter", BigFormat.latestVersion), ksDir);
+        copySstablesFromTestData(String.format("legacy_%s_simple", BigFormat.latestVersion), ksDir, "legacy_tables");
+        copySstablesFromTestData(String.format("legacy_%s_simple_counter", BigFormat.latestVersion), ksDir, "legacy_tables");
+        copySstablesFromTestData(String.format("legacy_%s_clust", BigFormat.latestVersion), ksDir, "legacy_tables");
+        copySstablesFromTestData(String.format("legacy_%s_clust_counter", BigFormat.latestVersion), ksDir, "legacy_tables");
     }
 
-    public static void copySstablesFromTestData(String table, File ksDir) throws IOException
+    public static void copySstablesFromTestData(String table, File ksDir, String ks) throws IOException
```

Comment:
This adds a `ks` parameter instead of using `"legacy_tables"`. However, the passed `ks` argument is always `"legacy_tables"`, so the change seems unneeded. 

Perhaps we could replace the 21 repetitions of the string `"legacy_tables"` by a `static final` class constant, make this method `private` and remove the parameter:
```java
private static final String LEGACY_TABLES_KEYSPACE = "legacy_tables";
...
private static void copySstablesFromTestData(String table, File ksDir) throws IOException
{
    File cfDir = new File(ksDir, table);
    cfDir.tryCreateDirectory();

    for (File srcDir : Keyspace.open(LEGACY_TABLES_KEYSPACE).getColumnFamilyStore(table).getDirectories().getCFDirectories())
    {
        for (File file : srcDir.tryList())
        {
            copyFile(cfDir, file);
        }
    }
}
```

Reply:
I like the static constant. But I would like to leave the method as it is. The new param allows other junits to 'capture' sstables whereas before it only worked for a specific KS. It's one of the changes I need to produce and capture corrupted sstables for the `CompactionsCQLTest` and to make the commented out sample code in there work. Wdyt?

Reply:
Makes sense to me, that would be used by the code that has been commented in `CompactionsCQLTest`. What are the other changes that we need?

Reply:
Sorry not following you now. We shouldn't need any further changes.

Reply:
@adelapena are we good here to resolve?

Reply:
I was referring to the part where you said above "It's one of the changes I need to produce and capture corrupted sstables for the `CompactionsCQLTest` and to make the commented out sample code in there work", and I was wondering what other things we need to make that commented code work.

Reply:
I think that's all. It was my idea to provide that code you can uncomment ready for the next dev that might need it. Everything else needed I baked seamlessly into the existing code already, such as having KS as a param here.

Reply:
Ah, I see, makes sense. For more clarity, we could add to `CompactionsCQLTest#testCompactionInvalidTombstone` and `CompactionsCQLTest#testCompactionInvalidPartitionDeletion` a comment saying `To generate the sstables with corrupted data, run this commented code in some pre-c14227 branch that allows for negative ldts`. 

By the way, I see that the table we are reading, `negative-ldts-invalid-deletions-test/invalidtombstones` has more two more rows than what is written by the commented code, so it seems they haven't been generated only with it, have they? The same happens with `negative-ldts-invalid-deletions-test//invalid_partition_deletion`. Both tables contain the rows `(0, 0, 0), (1, 1, 1)` in addition to the tombstone mentioned in the commented code.

Reply:
I think the comment you're asking for is there [already](https://github.com/apache/cassandra/pull/1891/files#diff-dbf9d67e1f6060176920b0d2555d111ef2d0fc0c3c3a0bec8e49c907155194eeR331)?

The 'extra' rows you are seeing are being inserted by the '[prepare](https://github.com/apache/cassandra/pull/1891/files#diff-dbf9d67e1f6060176920b0d2555d111ef2d0fc0c3c3a0bec8e49c907155194eeR394)()'  methods iiuc.

Reply:
> I think the comment you're asking for is there [already](https://github.com/apache/cassandra/pull/1891/files#diff-dbf9d67e1f6060176920b0d2555d111ef2d0fc0c3c3a0bec8e49c907155194eeR331)?

That link sends me to the global diff of all files, so I'm not sure what comment you are referring to. If it's [this one|https://github.com/bereng/cassandra/blob/CASSANDRA-14227-trunk/test/unit/org/apache/cassandra/db/compaction/CompactionsCQLTest.java#L331-L332], the part that I'd change is explicitly mentioning that the commented code generates the sstable. As in `To generate the sstables with corrupted data, run this commented code in some pre-c14227 branch that allows for negative ldts`.

> The 'extra' rows you are seeing are being inserted by the '[prepare](https://github.com/apache/cassandra/pull/1891/files#diff-dbf9d67e1f6060176920b0d2555d111ef2d0fc0c3c3a0bec8e49c907155194eeR394)()' methods iiuc.

Oh, right, I missed that.

Reply:
Will reword comment